bundles/test/org.eclipse.smarthome.test/src/main/java/org/eclipse/smarthome/test/OSGiTest.java

```diff
@@ -427,12 +428,109 @@ protected void setDefaultLocale(Locale locale) throws Exception {
         properties.put("variant", locale.getVariant());
 
         config.update(properties);
-        Closure closure = new Closure(null) {
-            public Object doCall() {
-                assertThat(localeProvider.getLocale(), is(locale));
-                return null;
+        waitForAssert(() -> assertThat(localeProvider.getLocale(), is(locale)));
+    }
+
+    /**
+     * When this method is called it waits until the condition is fulfilled or the timeout is reached.
+     * The condition is specified by a closure, that must return a boolean object. When the condition is
+     * not fulfilled Thread.sleep is called at the current Thread for a specified time. After this time
+     * the condition is checked again. By a default the specified sleep time is 50 ms. The default timeout
+     * is 10000 ms.
+     *
+     * @param condition closure that must not have an argument and must return a boolean value
+     * @param timeout timeout, default is 10000ms
+     * @param sleepTime interval for checking the condition, default is 50ms
+     * @throws Exception
+     */
+    @Deprecated
+    protected void waitFor(Closure<?> condition, int timeout, int sleepTime) throws Exception {
+        int waitingTime = 0;
+        while (condition != null && !Boolean.TRUE.equals(condition.call()) && waitingTime < timeout) {
```

Comment:
Yes, really bad. Just like the missing final assertion in `waitForAssert` which lead to successful tests after timeout.
wrt the `waitFor` methods: In the ESH codebase I only see the `JavaOSGiTest#waitFor` method used once and in a fashion where it could easily be migrated to `waitForAssert`. My suggestion is to deprecate all `waitFor` methods.